pkg/gitnessclient/gitness_client.go

```diff
@@ -0,0 +1,416 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package gitnessclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+const personalNamespaceId = "<PERSONAL>"
+
+type GitnessClient struct {
+	Token   string
+	BaseURL *url.URL
+}
+
+func NewGitnessClient(token string, baseUrl *url.URL) *GitnessClient {
+	return &GitnessClient{
+		Token: token,
+		BaseURL: &url.URL{
+			Scheme: "http",
+			Host:   baseUrl.Host,
+			Path:   baseUrl.Path,
+		},
+	}
+}
+
+func (g *GitnessClient) GetSpaceAdmin(spaceName string) (*apiSpaceMemberResponse, error) {
+
+	spacesURL, err := g.BaseURL.Parse(fmt.Sprintf("/api/v1/spaces/%s/members", spaceName))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url : %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), "GET", spacesURL.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Authorization", "Bearer "+g.Token)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to perform request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var apiMemberships []apiSpaceMemberResponse
+	if err := json.Unmarshal(body, &apiMemberships); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
+	}
+	var admin *apiSpaceMemberResponse
+	for _, member := range apiMemberships {
+		if member.Role == "space_owner" {
+			admin = &member
+			break
+		}
+	}
+	if admin == nil {
+		return nil, fmt.Errorf("failed to unmarshal response body: %w", err)
+	}
+	return admin, nil
```

Comment:
Remove the extra newline from the bottom as well. (You can move this return line down and leave an extra newline above if you wish.